### PR TITLE
Validate chunks from disk in availability-recovery

### LINKED ISSUE #6070: Check validity of chunks fetched from disk
Otherwise in case of a database corruption we would raise a dispute, although nobody did anything wrong (and we would get slashed).

This happened in the wild:

```
Data recovery invalid for candidate (0xdda32ca1be4330913b93a701a4c84192b8934764e8ae0f13e72518d5e4e97357, Id(2124)) para_id=Id(2124) candidate_hash=0xdda32ca1be4330913b93a701a4c84192b8934764e8ae0f13e72518d5e4e97357 traceID=294606636697047689533880607601836179858\n","stream":"stderr","time":"2022-09-21T19:18:38.194880785Z"}
```

-> Also check for other occurrences, where we would accuse others in case of a hardware error/local problem.

### PR DESCRIPTION
**Pull request by eskimor**
Otherwise we would trigger disputes on db corruptions.

Resolves #6070

### CODE REVIEW

**Review by ordian — APPROVED** [review]
Looks good.

We could also store erasure root in av-store directly and perform these checks on retrieval in `QueryChunk` and `QueryAllChunks`.

**Comment by eskimor** [comment]
Yep, I was thinking something [similar](https://github.com/paritytech/polkadot/issues/6076).

File: node/network/availability-recovery/src/lib.rs [hunk 1]
```diff
@@ -376,49 +376,20 @@ impl RequestChunksFromValidators {
...
-							?validator_index,
-							"Invalid Merkle proof",
+							validator_index = ?chunk.index,
+							"Received valid chunk",
 						);
-					}
-				},
+						self.received_chunks.insert(chunk.index, chunk);
+					} else {
+						metrics.on_chunk_request_invalid();
```
Comment (sandreim) [inline node/network/availability-recovery/src/lib.rs:390]: nit: we could also log a warning or debug message when an invalid chunk is received.

**Review by sandreim — APPROVED** [review]

File: node/network/availability-recovery/src/lib.rs (hunk 1 above)
Comment (eskimor) [reply node/network/availability-recovery/src/lib.rs:390]: We do - the `is_chunk_valid` function does.

**Comment by burdges** [comment]
Importantly, we should never dispute if received chunks fail their own Merkle proof check into the chunk root in the candidate receipt, but we do dispute if chunks reconstructed by erasure coding yield a different chunk root than what the candidate receipt says.  We should therefore never do reconstruction from chunks without first checking those chunks own Merkle proofs.

**Pull request merged**